src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -202,16 +295,25 @@ grpc_channel_credentials* grpc_google_default_credentials_create(void) {
   error = grpc_error_add_child(error, err);
 
   gpr_mu_lock(&g_state_mu);
-  /* At last try to see if we're on compute engine (do the detection only once
-     since it requires a network test). */
-  if (!g_compute_engine_detection_done) {
-    g_need_compute_engine_creds = g_gce_tenancy_checker();
-    g_compute_engine_detection_done = 1;
+
+  /* Try a platform-provided hint for GCE. */
+  if (!g_metadata_server_detection_done) {
+    g_is_on_gce = g_gce_tenancy_checker();
+    g_metadata_server_detection_done = g_is_on_gce;
```

Comment:
The two g_metadata_server variables are redundant: Whenever detection_done is true, available is true, and they only differ after flush(), at which time we'll overwrite available before the next read. Should we keep only g_metadata_server_available?

Reply:
Good catch. I removed the redundancy by using `g_metadata_server_available` only.